src/CodeGen.fs

```diff
@@ -1072,40 +1076,130 @@ let sampleFSharpClientMember query queryName hasVariables = stringBuffer {
+                    return Ok response.data
+
+            | errorStatus ->
+                let response = JsonSerializer.Deserialize<GraphqlErrorResponse> (responseContent, options)
+
+                return Error response.errors
+        }
+
+    member this.%s(%s) = Async.RunSynchronously(this.%sAsync%s)
+"""
+      queryName
+      (if hasVariables then "input: " + queryName + ".InputVariables" else "")
+      ("\"\"\"\n" + addLines query + "\n            \"\"\"")
+      (if hasVariables then "{ query = query; variables = Some input }" else "{ query = query; variables = None }")
+      queryName
+
+      queryName
+      (if hasVariables then "input: " + queryName + ".InputVariables" else "")
+      queryName
+      (if hasVariables then " input" else "()")
+}
+
+let inline sampleFSharpClientMember serializer query queryName hasVariables =
+        match serializer with
+        | SerializerType.System -> sampleFSharpSystemClientMember query queryName hasVariables
+        | SerializerType.Newtonsoft -> sampleFSharpNewtonsoftClientMember query queryName hasVariables
+        | SerializerType.Unknown -> sampleFSharpNewtonsoftClientMember query queryName hasVariables
+        | _ -> sampleFSharpNewtonsoftClientMember query queryName hasVariables
+
+let sampleFableGraphqlClient projectName clientName errorType members serializer = stringBuffer {
+        sprintf """namespace %s
 
 open Fable.SimpleHttp
 open Fable.SimpleJson
+open Newtonsoft.Json
+open Newtonsoft.Json.Linq
 
 type GraphqlInput<'T> = { query: string; variables: Option<'T> }
 type GraphqlSuccessResponse<'T> = { data: 'T }
 type GraphqlErrorResponse = { errors: %s list }
 
-type %s(url: string, headers: Header list) =
-    new(url: string) = %s(url, [ ])
+type %s(url: string, headers: Header list, settings: JsonSerializerSettings) =
+    new(url: string, settings: JsonSerializerSettings) = %s(url, [ ], settings)
+    new(url: string) = %s(url, [ ], new JsonSerializerSettings ())
 
-%s""" projectName errorType clientName clientName members
-}
+%s""" projectName errorType clientName clientName clientName members
+    }
 
-let sampleFSharpGraphqlClient projectName clientName errorType members = stringBuffer {
-    sprintf """namespace %s
+let private sampleFSharpSystemGraphqlClient projectName clientName errorType members =
+    stringBuffer {
+        sprintf """namespace %s
+
+open System.Net.Http
+open System.Text
+open System.Text.Json
+
+type GraphqlInput<'T> = { query: string; variables: Option<'T> }
+type GraphqlSuccessResponse<'T> = { data: 'T }
+type GraphqlErrorResponse = { errors: %s list }
+
+type %s(url: string, httpClient: HttpClient, options: JsonSerializerOptions) =
+    new(url: string, options: JsonSerializerOptions) = %s(url, new HttpClient(), options)
+    new(url: string) = %s(url, new HttpClient(), new JsonSerializerOptions())
+
+    %s""" projectName errorType clientName clientName clientName members
+    }
```

Comment:
Try using string interpolation. I suppose we don't need stringBuffer here.
It is better to cache values via let prior to string interpolation.